### PR TITLE
Dependency Update, Basic JWT support and more

### PR DESCRIPTION
**Pull request by paulausz**
# Fixes / Improvements
## Dependencies
- Updated dependencies to allow building the app using the Rocket.Chat App Engine CLI as discussed in #9 
- Added dependency for node to make is easier to build the app using the Rocket.Chat App Engine CLI

## README
- Added content to the README file to support users

## Language support
- Added German language support

# New Features
## Basic Json web token support
- Added an app setting to set a fix JWT access token as discussed in #5. A static JWT token is added to each conference link generated by the app if the setting is not empty. 
NewSetting

# TODO
- Please verify that the app can be deployed using the Markedplace. I haven't tried that.
- Please double check the README and the English language file for typos and grammatical mistakes
- The version number has not been increased yet

# Future work
I would like to create another pull request in the upcomming days to add more settings:
- Option to create random room names for channels
- Option to create random rooms names for direct messages
- Option to set the jitsi url directly in the app

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,13 @@
...
+`rc-apps deploy --url http://localhost:3000 --username <username> --password <password>`
+## Usage
+This app provides a Slashcommand. Simply type `/jitsi` in any message window and send the message. You can set a custom room name by adding the @ symbol to the message `/jitsi @MyRoomExample`. Please note that this app uses the Jitsi server specified in your Rocket.Chat Video settings: `Administration - Video Conference - Jitsi`
+## Settings
+The following settings can be set via `Administration - Apps - Jitsi Slash Command`
+### JWT Token
+When authentification via JSON Web Tokens (JWT) is enabled on your Jitsi instance, you may want to add a static token for each conference link generated to allow users to enter the conference. This token can be generated here: [jwt.io](https://jwt.io/). It is recommended so set an experiry date when creating the token. The token must not be limied to a dedicated room as the same token is added to very single conference link.
```
Comment (graywolf336) [inline README.md:12]:
```suggestion
When authentication via JSON Web Tokens (JWT) is enabled on your Jitsi instance, you may want to add a static token for each conference link generated to allow users to enter the conference. This token can be generated here: [jwt.io](https://jwt.io/). It is recommended to set an expiration date when creating the token. The token can not be limited to a dedicated room as the same token is added to every conference link. 
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,13 @@
 # Apps.Jitsi
 Rocket.Chat App for Jitsi
+
+## Installation
+Install the app via the Rocket.Chat Marketplace or download the source code and build the app by yourself. You would need the [Rpcket.Chat App Engine CLI](https://docs.rocket.chat/apps-development/getting-started/rocket.chat-app-engine-cli) to do so. To deploy a self-build app to your server, make sure to have the App development mode enabled (`Administration - General - Apps - Enable develpment mode`). Use the following command to deploy the app:
+`rc-apps deploy --url http://localhost:3000 --username <username> --password <password>`
+## Usage
```
Comment (graywolf336) [inline README.md:7]:
```suggestion

## Usage
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,13 @@
...
+This app provides a Slashcommand. Simply type `/jitsi` in any message window and send the message. You can set a custom room name by adding the @ symbol to the message `/jitsi @MyRoomExample`. Please note that this app uses the Jitsi server specified in your Rocket.Chat Video settings: `Administration - Video Conference - Jitsi`
+## Settings
```
Comment (graywolf336) [inline README.md:9]:
```suggestion

## Settings
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,13 @@
...
+The following settings can be set via `Administration - Apps - Jitsi Slash Command`
+### JWT Token
```
Comment (graywolf336) [inline README.md:11]:
```suggestion

### JWT Token
```

File: i18n/en.json [hunk 5]
```diff
@@ -1,4 +1,6 @@
 {
     "params_example": "[room name (optional)]",
-    "command_description": "Generate an Jitsi video meeting URL."
+    "command_description": "Generate an Jitsi video meeting URL.",
```
Comment (graywolf336) [inline i18n/en.json:3]:
```suggestion
    "command_description": "Generate a Jitsi video meeting URL.",
```

File: i18n/en.json [hunk 6]
```diff
@@ -1,4 +1,6 @@
...
+    "jwt_label" : "JWT Token",
+    "jwt_description": "Valid JWT Token to add to all conference links. Keep empty if you do not use JWT or you do not want to append a token when creating the link!"
```
Comment (graywolf336) [inline i18n/en.json:5]:
```suggestion
    "jwt_description": "Valid JWT Token which is added to all of the conference links. Keep it empty if you do not use JWT or you do not want to append a token when creating the link!"
```

File: index.ts [hunk 7]
```diff
@@ -1,10 +1,22 @@
...
 import { JitsiSlashCommand } from './slashcommand';
 
 export class JitsiApp extends App {
     protected async extendConfiguration(configuration: IConfigurationExtend): Promise<void> {
+        // Setting to add a JWT token
+        await configuration.settings.provideSetting({
+            id: 'jitsi_jwt',
+            type: SettingType.STRING,
+            packageValue: '',
+            required: true,
```
Comment (graywolf336) [inline index.ts:14]:
```suggestion
            required: false,
```

**Review by graywolf336 — CHANGES_REQUESTED** [review]
Awesome, thank you for submitting this PR and these changes! Here are a few things that are mostly nitpicky 🙈

**Comment by paulausz** [comment]
Thanks for your review @graywolf336. Have you also had a look at the other todos I mentioned?

**Comment by graywolf336** [comment]
No problem. I have not yet, I will be able to do those tomorrow and I'll report back to you. :)

**Comment by himpierre** [comment]
Hi Devs.

Self created app packages throws error.

- git clone
- git checkout v2
- rc-apps create
- rc-apps package
- install via Web
- Installs but with error -> "Resource not found"

Log:
`
{"level":50,"time":"2022-06-16T06:05:07.342Z","pid":15340,"hostname":"pages","name":"Rocket.Chat Apps","msg":"Unable to access Marketplace. Does the server has access to the internet? {\"code\":428,\"requestId\":\"51b82a82-f24c-4664-8fe4-f10a9ed86ec0\",\"status\":404}"} I20220616-08:07:31.319(2) FileStream {   _readableState: ReadableState {     objectMode: false,     highWaterMark: 16384,     buffer: BufferList { head: null, tail: null, length: 0 },     length: 0,     pipes: [],     flowing: null,     ended: false,     endEmitted: false,     reading: false,     sync: true,     needReadable: false,     emittedReadable: false,     readableListening: false,     resumeScheduled: false,     errorEmitted: false,     emitClose: true,     autoDestroy: true,     destroyed: false,     errored: null,     closed: false,     closeEmitted: false,     defaultEncoding: 'utf8',     awaitDrainWriters: null,     multiAwaitDrain: false,     readingMore: false,     dataEmitted: false,     decoder: null,     encoding: null,     [Symbol(kPaused)]: null   },   _events: [Object: null prototype] {     end: [Function: bound onceWrapper] { listener: [Function (anonymous)] }   },   _eventsCount: 1,   _maxListeners: undefined,   truncated: false,   _readcb: null,   [Symbol(kCapture)]: false } {"level":50,"time":"2022-06-16T06:07:31.980Z","pid":15340,"hostname":"pages","name":"Rocket.Chat Apps","msg":"Unable to access Marketplace. Does the server has access to the internet? {\"code\":428,\"requestId\":\"03b1efe8-cb0e-478e-a677-7b7249ec482d\",\"status\":404}"} 
`

Server has for sure internet access.

cheers, t.

**Comment by pierre-lehnen-rc** [comment]
I think that's an unrelated error on the rocket.chat side that doesn't prevent the app from working. 

But jitsi app v2 also relies on code that is not yet merged on rocket.chat (will be merged next week and released on version 5.0).

**Comment by himpierre** [comment]
Yay!

Am 16. Juni 2022 08:28:04 MESZ schrieb Pierre Lehnen ***@***.***>:
>I think that's an unrelated error on the rocket.chat side that doesn't prevent the app from working. 
>
>But jitsi app v2 also relies on code that is not yet merged on rocket.chat (will be merged next week and released on version 5.0). 
>

**Comment by pierre-lehnen-rc** [comment]
I'm implementing token generation on the app as well. Should I still keep the option to use a static token or it won't be needed anymore?

**Pull request merged**